### PR TITLE
Adi/jooq api

### PR DESCRIPTION
**Pull request by AdiFinkelman**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -32,7 +32,7 @@ dependencies {
 	implementation("org.flywaydb:flyway-core")
 	implementation("org.flywaydb:flyway-database-postgresql")
 	implementation("org.jetbrains.kotlin:kotlin-reflect")
-	implementation("org.jooq:jooq:3.16.9")
+	implementation("org.jooq:jooq:3.17.9")
```
Comment (omermelamed) [inline build.gradle.kts:35]: Lets separate this version update to a different pr.

File: src/main/kotlin/com/hibob/academy/resource/OwnerApi.kt [hunk 2]
```diff
@@ -11,13 +13,14 @@ import java.util.concurrent.CopyOnWriteArrayList
...
-class OwnerResource {
+class OwnerResource(private val ownerService: OwnerService) {
 
     private val allOwners: MutableList<Owner> = CopyOnWriteArrayList()
+    private val companyId = 2L
 
     @GET
     @Path("/{ownerId}")
-    fun getOwnerById(@PathParam("ownerId") ownerId: Long): Response {
+    fun getOwnerById(@PathParam("ownerId") ownerId: Long?): Response {
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/resource/OwnerApi.kt:23]: OwnerId can be nullable?
If we want to get the owner by id then shouldn't be nullable

File: src/main/kotlin/com/hibob/academy/resource/OwnerApi.kt [hunk 3]
```diff
@@ -68,17 +80,20 @@ class OwnerResource {
...
 
-    private fun updateFirstAndLastName(updatedOwner: Owner): Pair<String?, String?>  {
-        val firstName = updatedOwner.name.split(" ").firstOrNull() ?: updatedOwner.firstName ?: throw BadRequestException("First name not found")
-        val lastName = updatedOwner.name.split(" ").drop(1).joinToString(" ").takeIf { it.isNotBlank() } ?: updatedOwner.lastName ?: throw BadRequestException("Last name not found")
+    private fun updateFirstAndLastName(updatedOwner: Owner): Pair<String?, String?> {
+        val firstName = updatedOwner.name.split(" ").firstOrNull() ?: updatedOwner.firstName
+        ?: throw BadRequestException("First name not found")
+        val lastName =
+            updatedOwner.name.split(" ").drop(1).joinToString(" ").takeIf { it.isNotBlank() } ?: updatedOwner.lastName
+            ?: throw BadRequestException("Last name not found")
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/resource/OwnerApi.kt:96]: Nice

File: src/main/kotlin/com/hibob/academy/resource/PetApi.kt [hunk 4]
```diff
@@ -30,20 +33,29 @@ class PetsResource {
...
         return Response.ok()
-            .entity(newPet)
+            .entity(petToAdd)
             .build()
     }
 
+    //jooq task
+    @PUT
+    @Path("/adopt")
+    fun adoptPet(adoptionRequest: AdoptionRequest): Response {
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/resource/PetApi.kt:52]: Nice, great work on using consumes.

**Review by omermelamed — COMMENTED** [review]
Well done, please add tests to the service.
Left few comments.

File: src/main/kotlin/com/hibob/academy/resource/PetApi.kt (hunk 4 above)
Comment (AdiFinkelman) [reply src/main/kotlin/com/hibob/academy/resource/PetApi.kt:52]: Thanks

**Review by omermelamed — APPROVED** [review]
Great job 👏 well written pr.
Keep it that way
